aten/src/ATen/core/ivalue_inl.h

```diff
@@ -290,18 +290,22 @@ struct C10_EXPORT ivalue::Future : c10::intrusive_ptr_target {
   /**
    * Wait on the future until it completes.
    */
-  virtual void wait() {
+  void wait() {
     std::unique_lock<std::mutex> lock(mutex_);
     while (!completed_) {
       finished_cv_.wait(lock);
     }
+
+    if (!eptr_) {
```

Comment:
shall we use `hasError` instead?

Reply:
I don't think we can use it as-is, since it tries to acquire the mutex, and this method already has it, and that would deadlock. If you think it's worth it I can add a `hasErrorInternal` method that doesn't acquire the mutex.